src/lib/optimizer/chunk_statistics.hpp

```diff
@@ -0,0 +1,171 @@
+#pragma once
+
+#include <exception>
+#include <memory>
+#include <type_traits>
+#include <vector>
+
+#include "all_type_variant.hpp"
+#include "type_cast.hpp"
+#include "types.hpp"
+
+#include "storage/base_encoded_column.hpp"
+#include "storage/deprecated_dictionary_column.hpp"
+#include "storage/dictionary_column.hpp"
+#include "storage/reference_column.hpp"
+#include "storage/run_length_column.hpp"
+#include "storage/value_column.hpp"
+
+namespace opossum {
+
+// select how many ranges we want in the filter
+// make this customizable?
+static constexpr uint32_t MAX_RANGES_COUNT = 10;
+
+class BaseFilter : public std::enable_shared_from_this<BaseFilter> {
```

Comment:
Bit of doc would be helpful. What do derived classes do, where is this used?

Reply:
Also this falls out of the one class/one file scheme, please put it into its own file